### PR TITLE
Add ClojureCLR support

### PR DESCRIPTION
**Pull request by E-A-Griffin**
Added support for ClojureCLR repl via `lein-clr`. Updated README to reflect changes.

### CODE REVIEW

File: inf-clojure.el [hunk 1]
```diff
@@ -75,6 +75,7 @@
                                     (boot . "boot repl")
                                     (clojure . "clojure")
                                     (cljs . "clojure -m cljs.main -r")
+                                    (cljr . "lein clr repl")
```
Comment (bbatsov) [inline inf-clojure.el:78]: Might be better to name this `lein-cljr` in case we decide to add more ways to start a cljr repl down the road.

**Comment by bbatsov** [comment]
Please, mention this in the changelog as well.

**Comment by bbatsov** [comment]
Btw, I've also noticed that `lein-clr` hasn't been updated in 7 years, but I guess it still gets the job done.

**Comment by E-A-Griffin** [comment]
> Btw, I've also noticed that `lein-clr` hasn't been updated in 7 years, but I guess it still gets the job done.

Yeah I noticed that as well. There's at least one feature I think it should be updated to include (dealing with dependency management for Clojure/ClojureScript/ClojureCLR projects where some dependencies aren't compatible with all three variants of Clojure). I may reach out to the repo owner and see if it would be appropriate to move the repo to something like https://github.com/clj-commons. I'd be willing to try to contribute to its maintenance though I wouldn't want to take on that responsibility solely.

File: inf-clojure.el (hunk 1 above)
Comment (E-A-Griffin) [reply inf-clojure.el:78]: I completely agree, especially with the hope that more modern tools for `cljr` repls become available soon.

**Comment by bbatsov** [comment]
Sounds like a good idea to move the project to `clj-commons`.

**Comment by E-A-Griffin** [comment]
Let me know if these changes address all concerns or if further changes are still needed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -123,3 +123,4 @@
 ### New Features
 
 * [#22](https://github.com/clojure-emacs/inf-clojure/pull/22): Add ElDoc support.
+* [#202](https://github.com/clojure-emacs/inf-clojure/issues/202): Add ClojureCLR support.
```
Comment (bbatsov) [inline CHANGELOG.md:126]: That should be under the `master` section.

**Comment by bbatsov** [comment]
@E-A-Griffin Just move the changelog entry and we're good to go.

**Comment by E-A-Griffin** [comment]
> @E-A-Griffin Just move the changelog entry and we're good to go.

Awesome, should be good now then.

**Pull request merged**